src/main/java/gyro/google/compute/InstanceResource.java

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright 2019, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.compute;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.model.Instance;
+import com.google.api.services.compute.model.InstancesSetLabelsRequest;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Regex;
+import gyro.core.validation.Required;
+import gyro.google.Copyable;
+
+/**
+ * Creates an instance.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *      zone: "us-west1-a"
+ *
+ *      google::instance gyro-dev-1
+ *          name: "gyro-development"
+ *          description: "Testing for Gyro"
+ *          zone: $zone
+ *          machine-type: "zones/$(zone)/machineTypes/n1-standard-1"
+ *
+ *          network-interfaces
+ *              network: $(external-query google::compute-network {name: "default"})
+ *          end
+ *
+ *          initialize-disks
+ *              boot: true
+ *
+ *              initialize-params
+ *                  disk-name: "gyro-boot-disk"
+ *                  source-image: "projects/debian-cloud/global/images/family/debian-9"
+ *              end
+ *          end
+ *
+ *          initialize-disks
+ *              initialize-params
+ *                  disk-name: "gyro-secondary-disk"
+ *                  source-image: "projects/debian-cloud/global/images/family/debian-9"
+ *              end
+ *          end
+ *
+ *          labels: {
+ *              "gyro": "install"
+ *          }
+ *      end
+ */
+@Type("instance")
```

Comment:
Prefix this with `compute-` and update the examples with the new type:
```suggestion
@Type("compute-instance")
```

Reply:
Done.